url82787aa7a4_vrthra_v_language_tgz-pJ8-v_VJ8/src/v/java/Helper.java

```
@@ -0,0 +1,212 @@
+    }
+
+    public static Field getField(String c, String method) {
+        try {
+            Class cls = getClass(c);
+            return cls.getField(method.substring(0, method.length() - 1));
+        } catch (NoSuchFieldException e) {
+            throw new VException("err:java:no_such_field",new Term<String>(v.Type.TString,method), "Java Exception : " + e.getMessage());
+        }
+    }
+
+    public static Term invoke(Term c, Term m, Quote params) {
+        String cname = null;
+        Object o = null;
+        String method = m.svalue();
+        if (c.type != v.Type.TSymbol) {
+            o = getJavaObj(c);
+            cname = o.getClass().getName();
+        } else {
+            cname = c.svalue();
+            o = null; // static invocation.
+        }
+
+        try {
+            if (method.equals("new")) {
+                if (o != null)
+                    throw new VException("err:java:invalid_class_name",c, "Java Exception invalid class");
+                Constructor cons = getConstructor(cname, params);
+                Object[] args = getParams(params);
+                return getResult(cons.newInstance(args));
+            } else if (method.endsWith("$")) {
+                // TODO: static field.
+                // Field.
+                Field fld = getField(cname, method);
+                // is it get or set?
+                int size = ((QuoteStream)params.tokens()).size();
+
+                if (size > 0) { // set
+                    Object no = getJavaObj(((QuoteStream)params.tokens()).get(0));
+                    fld.set(o, no);
+                    return getResult(no);
+                } else { // get
+                    return getResult(fld.get(o));
+                }
+            } else {
+                Method mtd = getMethod(cname, m, params);
+                Object[] args = getParams(params);
+                return getResult(mtd.invoke(o, args));
+            }
+        } catch (IllegalAccessException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 206, 208.